src/test/java/webpayplus/OneclickMallDeferredTest.java

```diff
@@ -0,0 +1,304 @@
+        setResponsePut(url, jsonResponse);
+
+        final OneclickMallInscriptionFinishResponse response = Oneclick.MallDeferredInscription.finish(testToken);
+        assertEquals(response.getResponseCode(), responseCode);
+        assertEquals(response.getTbkUser(), tbkUser);
+        assertEquals(response.getAuthorizationCode(), authorizationCode);
+        assertEquals(response.getCardType(), cardType);
+        assertEquals(response.getCardNumber(), cardNumber);
+
+    }
+
+
+     private Map<String, Object> generateAutorizeJsonResponse(){
+
+        Map<String, String> mapResponseCardDetail = new HashMap<String, String>();
+        mapResponseCardDetail.put("card_number", cardNumber);
+
+        Map<String, Object> mapResponseDetail1 = new HashMap<String, Object>();
+        mapResponseDetail1.put("amount", amount1);
+        mapResponseDetail1.put("status", status1);
+        mapResponseDetail1.put("authorization_code", authorizationCode1);
+        mapResponseDetail1.put("payment_type_code", paymentTypeCode1);
+        mapResponseDetail1.put("response_code", responseCode1);
+        mapResponseDetail1.put("installments_number", installmentsNumber1);
+        mapResponseDetail1.put("commerce_code", commerceCode1);
+        mapResponseDetail1.put("buy_order", buyOrder1);
+
+        Map<String, Object> mapResponseDetail2 = new HashMap<String, Object>();
+        mapResponseDetail2.put("amount", amount2);
+        mapResponseDetail2.put("status", status2);
+        mapResponseDetail2.put("authorization_code", authorizationCode2);
+        mapResponseDetail2.put("payment_type_code", paymentTypeCode2);
+        mapResponseDetail2.put("response_code", responseCode2);
+        mapResponseDetail2.put("installments_number", installmentsNumber2);
+        mapResponseDetail2.put("commerce_code", commerceCode2);
+        mapResponseDetail2.put("buy_order", buyOrder2);
+
+
+        Map<String, Object> mapResponse = new HashMap<String, Object>();
+        mapResponse.put("buy_order", buyOrder);
+        mapResponse.put("card_detail", mapResponseCardDetail);
+        mapResponse.put("accounting_date", accountingDate);
+        mapResponse.put("transaction_date", transactionDate);
+        //details
+        List<Map<String, Object>> lstDetail = new ArrayList<Map<String, Object>>();
+        lstDetail.add(mapResponseDetail1);
+        lstDetail.add(mapResponseDetail2);
+        mapResponse.put("details", lstDetail);
+
+        return mapResponse;
+    }
+
+    @Test
+    public void authorize() throws IOException, TransactionAuthorizeException {
+        Oneclick.setIntegrationType(IntegrationType.SERVER_MOCK);
+        String url = String.format("%s/transactions",apiUrl);
+
+        Map<String, Object> mapResponse = generateAutorizeJsonResponse();
+        Gson gson = new GsonBuilder().create();
+        setResponsePost(url, gson.toJson(mapResponse));
+
+        String tbkUserReq = "12350bde-00dd-4ad8-9cc6-ae918022adc3";
```

Comment:
I know these are integration tokens and everyone has access to them but we can still use completely fake ones and tests will be ok. Something like 'aaaaaaaaaaaaabbbbbbbbcccccc'